### PR TITLE
last stdlib cleanups

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -67,6 +67,31 @@ type
...
+  - browsers
+  - collections/rtarrays
+  - collections/sharedlist
+  - collections/sharedtable
+  - concurrency/atomics
+  - concurrency/cpuload
+  - concurrency/threadpool
+  - coro
+  - endians
+  - httpcore
```
Comment (dom96) [inline changelog.md]: The problem with marking these as "unstable" is that all their dependencies are unstable too, so you should probably mark those as "unstable" too...

Does `httpcore`` really need to be "unstable"? I can understand ``asyncstreams`` but ugh, that makes `httpclient` unstable...

Comment (Araq) [reply changelog.md]: That's not how it works, the stability refers to the API and every client of ``httpcore`` that is under our control is safe and will be patched if change the API.

Comment (dom96) [reply changelog.md]: yes, but `httpclient` exposes both asyncstreams and httpcore types. It feels a little misleading to not treat them the same. But I guess it's technically correct, we should be able to keep the current functions that return the `asyncstreams` object and introduce a new function that returns some new thing that we implement without breaking backwards compat.

I don't understand what's wrong with httpcore though. Can you explain?

Comment (Araq) [reply changelog.md]: Issue https://github.com/nim-lang/Nim/issues/12211 implies keeping the API the way it is will be hard.

Comment (disruptek) [reply changelog.md]: I agree that it sucks to have to call it unstable, but I also agree that there are some behaviors that seem undesirable to support indefinitely.  Can we just add a few lines to fix `httpclient` to work around misbehaving `httpcore` input pre-1.0 and then post-1.0 just add a new type or two to produce more correct inputs for new code?  It's really 11th hour and I hate that, but this seems to meet both concerns...  :frowning_face:

Comment (Araq) [reply changelog.md]: @disruptek Go for it but I won't let it delay the release which is scheduled for Monday.

**Pull request merged**